gobblin-utility/src/main/java/gobblin/broker/iface/UnscopedSharedResourceFactory.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.broker.iface;
+
+/**
+ * A {@link SharedResourceFactory} that creates global objects (not specific to any {@link Scope}).
+ * @param <T> type of objects this {@link SharedResourceFactory} creates.
+ * @param <K> type of {@link SharedResourceKey} this factory uses.
+ */
+public interface UnscopedSharedResourceFactory<T, K extends SharedResourceKey> extends SharedResourceFactory<T, K> {
```

Comment:
Can you think if we can avoid the parallel class hierarchies for scoped and unscoped?